src/Infrastructure/LeanCode.AuditLogs/AzureBlobAuditLogStorage.cs

```diff
@@ -0,0 +1,129 @@
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure;
+using Azure.Data.Tables;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Serilog;
+
+namespace LeanCode.AuditLogs;
+
+public class AzureBlobAuditLogStorage : IAuditLogStorage
+{
+    private readonly ILogger logger = Log.ForContext<AzureBlobAuditLogStorage>();
+
+    private static ReadOnlySpan<byte> NewLineBytes => "\n"u8;
+    private static readonly JsonSerializerOptions Options =
+        new()
+        {
+            Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+            ReferenceHandler = ReferenceHandler.IgnoreCycles,
+            WriteIndented = false,
+        };
+
+    private readonly BlobServiceClient blobClient;
+    private readonly TableServiceClient tableClient;
+    private readonly AzureBlobAuditLogStorageConfiguration config;
+
+    public AzureBlobAuditLogStorage(
+        BlobServiceClient blobClient,
+        TableServiceClient tableClient,
+        AzureBlobAuditLogStorageConfiguration config
+    )
+    {
+        this.blobClient = blobClient;
+        this.tableClient = tableClient;
+        this.config = config;
+    }
+
+    public async Task StoreEventAsync(AuditLogMessage auditLogMessage, CancellationToken cancellationToken)
+    {
+        try
+        {
+            await AppendLogToBlobAsync(auditLogMessage, cancellationToken);
+        }
+        catch (RequestFailedException e)
+        {
+            if (e.ErrorCode == BlobErrorCode.BlockCountExceedsLimit)
+            {
+                await BumpSuffixInTableAsync(auditLogMessage, cancellationToken);
+                await AppendLogToBlobAsync(auditLogMessage, cancellationToken);
+            }
+            else
+            {
+                throw;
+            }
+        }
+    }
+
+    private async Task AppendLogToBlobAsync(AuditLogMessage auditLogMessage, CancellationToken cancellationToken)
+    {
```

Comment:
Let's refactor this (simply extracting methods) into:
1. `CreateBlobAsync` (that gets the `BlobClient` and calls `CreateIfNotExistAsync`)
2. Serialize (that serializes the data to memory stream that it then returns)
3. Append & log (just like it is now)

Basically let's make this a 4-line method.